### DISCUSSION TITLE
scroll-middle (scrolloff and scrollfix) for columns

### DISCUSSION THREAD

**Discussion by geekscrapy** [saulpw/visidata#1830] [Ideas]
Is there such a command to center a view on a particular column like `scroll-middle`?

Scenario is that you have a lot of columns and you want to put the selected column to the far left, or center it. This is currently not possible without moving across x number of columns so that the one you want is on the far left.

This idea is different to key columns as you are just wanting to _review_ the data in the columns, not have reference to the data in the keycol.

Another idea is like https://github.com/saulpw/visidata/issues/1441#issuecomment-1489985885 (`scrolloff` and `scrollfix`) but for columns

**Comment by saulpw** [comment]
No command yet!  I would love this too.  Scrolling columns has been tricky; pageLeft basically does it one column at a time and recalculates, to see what will fit.  We can either copy or refactor that code, or figure out a cleaner way to do it entirely.

**Reply by geekscrapy** [reply]
Sweet. Would this have any reference to the scrolloff (etc) plugin I put forward? Just thinking whether to leave the PR for scrolloff until someone wants to take a crack at this. As the code might be similar

**Reply by saulpw** [reply]
I think the scrolloff code you already have is basically ready for a PR.  No sense in waiting if you have working code ready!
